pkg/agent/attestor/node/node.go

```diff
@@ -221,25 +225,26 @@ func (a *attestor) readSVIDFromDisk() []*x509.Certificate {
 func (a *attestor) newSVID(ctx context.Context, key *ecdsa.PrivateKey, bundle *bundleutil.Bundle) (newSVID []*x509.Certificate, newBundle *bundleutil.Bundle, err error) {
 	counter := telemetry_agent.StartNodeAttestorNewSVIDCall(a.c.Metrics)
 	defer counter.Done(&err)
+	attestorName := new(string)
+	defer func() { telemetry_common.AddAttestorType(counter, *attestorName) }()
```

Comment:
Using a pointer is a little smelly to me. The variable is being captured so when the defer is executed, it will have the latest value of `attestorName`.

https://play.golang.org/p/i_hDjc1LfI1

I think this could be implemented more succintly by moving the call to `AddAttestorType` into a defer that happens right after the declaration of attestorName on line 229.